"Undefined reference to nio_*" errors despite linking against libnspireio.a

## Steps to Reproduce

1. Run `mkdir niotest && cd niotest && nspire-tools new niotest`
2. Save the following to `main.cpp`:
   ```cpp
   #include <nspireio/nspireio.h>

   int main(void) {
       nio_console console;
       nio_init(&console, NIO_MAX_COLS, NIO_MAX_ROWS, 0, 0, NIO_COLOR_BLACK, NIO_COLOR_WHITE, true);
       nio_set_default(&console);
       nio_printf("test\n");
   }
   ```
3. Find `LDFLAGS =` in `Makefile`, and append `-l:libnspireio.a` to it
4. `make`

## Expected behavior

It builds successfully

## Actual behavior

```shell
> make
nspire-g++ -Wall -W -marm -Os -c main.cpp -o main.o
mkdir -p .
nspire-ld main.o -o niotest.elf -l:libnspireio.a
arm-none-eabi-ld: main.o: in function `main':
main.cpp:(.text.startup+0x34): undefined reference to `nio_init(void**, int, int, int, int, unsigned char, unsigned char, bool)'
arm-none-eabi-ld: main.cpp:(.text.startup+0x3c): undefined reference to `nio_set_default(void**)'
arm-none-eabi-ld: main.cpp:(.text.startup+0x44): undefined reference to `nio_printf(char const*, ...)'
collect2: error: ld returned 1 exit status
make: *** [Makefile:40: niotest.elf] Error 1
```

## Miscellaneous information

libnspireio.a _does_ seem to have the functions:
```shell
> grep nio_init /usr/share/ndless-sdk/lib/libnspireio.a
grep: /usr/share/ndless-sdk/lib/libnspireio.a: binary file matches
> grep nio_set_default /usr/share/ndless-sdk/lib/libnspireio.a
grep: /usr/share/ndless-sdk/lib/libnspireio.a: binary file matches
> grep nio_printf /usr/share/ndless-sdk/lib/libnspireio.a
grep: /usr/share/ndless-sdk/lib/libnspireio.a: binary file matches
```

Also, I built and installed ndless-sdk via [the AUR package](https://aur.archlinux.org/packages/ndless-sdk-git)

--- a/include/nspireio/console.hpp
+++ b/include/nspireio/console.hpp
@@ -25,9 +25,7 @@
  * C++ specific console header
  */
 
-extern "C" {
-	#include "nspireio.h"
-}
+#include "nspireio.h"
 
 #include "ios_base.ipp"
 

--- a/include/nspireio/nspireio.h
+++ b/include/nspireio/nspireio.h
@@ -28,6 +28,10 @@
 #ifndef NSPIREIO_H
 #define NSPIREIO_H
 
+#ifdef __cplusplus
+extern "C" {
+#endif
+
 #include "platform.h"
 #include "queue.h"
 
@@ -492,6 +496,10 @@ void nio_cursor_custom(nio_console* c, unsigned char cursor_data[6]);
 #define nio_SetCursorType				nio_cursor_type
 #define nio_SetCursorWidth				nio_cursor_width
 #define nio_SetCursorCustom				nio_cursor_custom
+#endif
+
+#ifdef __cplusplus
+}
 #endif
 
 #endif

--- a/include/nspireio/uart.hpp
+++ b/include/nspireio/uart.hpp
@@ -25,9 +25,7 @@
  * C++ specific UART header
  */
 
-extern "C" {
-	#include "nspireio.h"
-}
+#include "nspireio.h"
 
 #include "ios_base.ipp"
 

